### PR TITLE
`Lectures`: Fix file copying for attachment import

### PR DESCRIPTION
**Pull request by JohannesWt**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).

### Motivation and Context

Currently there is an issue when importing a lecture the attachments of the imported lecture unit are not transferred. Solves #9059 

### Description

Attachment Units of a lecture have not been copied correctly. Now there is a new implementation for copying attachments units as well as attachments directly linked to a lecture.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 course with 1 Lecture having
    - 1 Direct Attachment
    - 1 Attachment Unit
- 1 empty course

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to the empty course to lectures
4. click on "Import lectures" and choose the lecture of your other course
5. Click on import
6. Open the imported lecture attachment and verify that can be opened 
7. Click on the "Download"-Button of the attachment unit and verify that the file can be opened/downloaded

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage

unchanged

## Summary by CodeRabbit

- **New Features**
	- Enhanced lecture import functionality with improved attachment management and modular file handling.

- **Bug Fixes**
	- Streamlined error handling during attachment cloning for greater reliability and robustness.

- **Tests**
	- Updated test cases to ensure attachments are contextually linked to lectures, enhancing test reliability.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/LectureImportService.java [hunk 1]
```diff
@@ -172,7 +173,7 @@ else if (importedLectureUnit instanceof ExerciseUnit) {
      * @param importedAttachment The original attachment to be copied
```
Comment (kaancayli) [inline src/main/java/de/tum/in/www1/artemis/service/LectureImportService.java:173]: The `@param` annotations in the comment can be extended to align with the current method signature.

**Review by kaancayli — CHANGES_REQUESTED** [review]
Tested on TS1. I was able to open the attachment and download lecture units. The functionality works as expected. 👍

I have left a minor comment for you that you might want to address.

**Review by pzdr7 — CHANGES_REQUESTED** [review]
One comment on the code

----

Tested on TS1. Import of units and attachments works fine, but I've noticed one edge case:

1. Create a new lecture and use the automatic unit processing.
2. Import the lecture.
3. Go e.g. to the communications view. Open a channel and, in the markdown editor, click on "Lecture > { the original lecture, the lecture you've imported } > any of the units".
4. The slides are available for the original lecture but not for the imported one.

Do we want to handle this case?

**Review by BBesrour — DISMISSED** [review]
code lgtm, except for the missing javadoc param as mentioned in previous comments

**Review by laadvo — APPROVED** [review]
Code looks good

**Review by kaancayli — APPROVED** [review]
Changes look good 👍🏻

**Review by BBesrour — APPROVED** [review]
re-approve code

**Review by krusche — APPROVED** [review]

**Pull request merged**